polygeist/lib/Dialect/Polygeist/Analysis/UniformityAnalysis.cpp

```diff
@@ -127,6 +127,24 @@ LogicalResult UniformityAnalysis::initialize(Operation *top) {
   return SparseDataFlowAnalysis::initialize(top);
 }
 
+void UniformityAnalysis::setToEntryState(UniformityLattice *lattice) {
+  // The arguments of a gpu kernel are uniform by default.
+  if (Region *region = lattice->getPoint().getParentRegion()) {
+    if (auto gpuFuncOp = region->getParentOfType<gpu::GPUFuncOp>()) {
+      if (gpuFuncOp.isKernel()) {
+        for (Value arg : gpuFuncOp.front().getArguments()) {
+          UniformityLattice *lattice = getLatticeElement(arg);
+          propagateIfChanged(lattice, lattice->join(Uniformity::getUniform()));
+        }
+        return;
+      }
+    }
+  }
+
+  // Other arguments have unknown uniformity.
+  propagateIfChanged(lattice, lattice->join(Uniformity::getUnknown()));
```

Comment:
is `setToEntryState` only invoked by function arguments?

Reply:
It is called by `initialize` on every argument to a Region.